bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt

```diff
@@ -507,6 +526,102 @@ private constructor(
         }
     }
 
+    fun stream(parameters: Parameters?, callback: WebSocketCallback): WebSocket {
+        val streamingUrl: HttpUrl = if (streamingUrl != null) {
+            // okhttp doesn't support ws/wss scheme, so we need to convert ws to http, wss to https
+            val isSecureScheme: Boolean = streamingUrl.startsWith("https")
+            val scheme: String = if (isSecureScheme) "https" else "http"
+            // Remove the scheme portion ( https:// or http:// ) from the full url
+            val instanceName: String = streamingUrl.substring(if (isSecureScheme) 8 else 7)
```

Comment:
I'm not sure if I understand this correctly. The first comment reads as if we're going to specifically remove "ws" and "wss" prefixes from `streamingUrl` - but for `isSecureScheme`, we check if there's a "https" prefix, and then claim that we're removing "http[s]://" prefixes, which also matches the number of characters being removed.

If this is correct, I think some of the comment should be rephrased to be less confusing. ;)

Also, would it be safer to go for `streamingUrl.removePrefix("https://")` (or whatever the correct string to remove ends up being), so that we don't remove arbitrary prefixes in the error case?

Reply:
I’ve rewritten that whole portion in [75be158](https://github.com/andregasser/bigbone/pull/383/commits/75be1583e693501828a964e9675f17fa67df6678) to now be more stable and less confusing. Thanks for pointing it out!